vendor/bat-native-ads/src/bat/ads/internal/database/tables/campaigns_database_table.cc

```diff
@@ -0,0 +1,157 @@
+    DBTransaction* transaction,
+    const CreativeAdList& creative_ads) {
+  DCHECK(transaction);
+
+  if (creative_ads.empty()) {
+    return;
+  }
+
+  DBCommandPtr command = DBCommand::New();
+  command->type = DBCommand::Type::RUN;
+  command->command = BuildInsertOrUpdateQuery(command.get(), creative_ads);
+
+  transaction->commands.push_back(std::move(command));
+}
+
+std::string Campaigns::get_table_name() const {
+  return kTableName;
+}
+
+void Campaigns::Migrate(
+    DBTransaction* transaction,
+    const int to_version) {
+  DCHECK(transaction);
+
+  switch (to_version) {
+    case 3: {
+      MigrateToV3(transaction);
+      break;
+    }
+
+    default: {
+      break;
+    }
+  }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+int Campaigns::BindParameters(
+    DBCommand* command,
+    const CreativeAdList& creative_ads) {
+  DCHECK(command);
+
+  int count = 0;
+
+  int index = 0;
+  for (const auto& creative_ad : creative_ads) {
+    BindString(command, index++, creative_ad.campaign_id);
+    BindInt64(command, index++, creative_ad.start_at_timestamp);
+    BindInt64(command, index++, creative_ad.end_at_timestamp);
+    BindInt(command, index++, creative_ad.daily_cap);
+    BindString(command, index++, creative_ad.advertiser_id);
+    BindInt(command, index++, creative_ad.priority);
+    BindDouble(command, index++, creative_ad.ptr);
+
+    count++;
+  }
+
+  return count;
+}
+
+std::string Campaigns::BuildInsertOrUpdateQuery(
+    DBCommand* command,
+    const CreativeAdList& creative_ads) {
+  const int count = BindParameters(command, creative_ads);
+
+  return base::StringPrintf(
+      "INSERT OR REPLACE INTO %s "
+          "(campaign_id, "
+          "start_at_timestamp, "
+          "end_at_timestamp, "
+          "daily_cap, "
+          "advertiser_id, "
+          "priority, "
+          "ptr) VALUES %s",
+      get_table_name().c_str(),
+      BuildBindingParameterPlaceholders(7, count).c_str());
+}
+
+void Campaigns::CreateTableV3(
+    DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  const std::string query = base::StringPrintf(
+      "CREATE TABLE %s "
+          "(campaign_id TEXT NOT NULL, "
+          "start_at_timestamp TIMESTAMP NOT NULL, "
+          "end_at_timestamp TIMESTAMP NOT NULL, "
+          "daily_cap INTEGER DEFAULT 0 NOT NULL, "
+          "advertiser_id TEXT NOT NULL, "
+          "priority INTEGER NOT NULL DEFAULT 0, "
+          "ptr DOUBLE NOT NULL DEFAULT 1, "
+          "PRIMARY KEY (campaign_id), "
+          "UNIQUE(campaign_id) ON CONFLICT REPLACE)",
+      get_table_name().c_str());
+
+  DBCommandPtr command = DBCommand::New();
+  command->type = DBCommand::Type::EXECUTE;
+  command->command = query;
+
+  transaction->commands.push_back(std::move(command));
+}
+
+void Campaigns::CreateIndexV3(
+    DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  CreateIndex(transaction, get_table_name(), "campaign_id");
```

Comment:
Nit: I don't think we need an index since `campaign_id` is already a PRIMARY KEY as mentioned in https://stackoverflow.com/questions/3379292/is-an-index-needed-for-a-primary-key-in-sqlite